bin/postgres-ha/bootstrap/pre-bootstrap.sh

```diff
@@ -322,6 +293,12 @@ build_bootstrap_config_file() {
         /opt/cpm/bin/yq m -i -x "${bootstrap_file}" "/opt/cpm/conf/postgres-ha-initdb.yaml"
     fi
 
+    if [[ -n "${PGHA_WALDIR}" ]]
+    then
+        echo_info "Applying custom WAL dir to postgres-ha configuration"
+        /opt/cpm/bin/yq w -i "${bootstrap_file}" 'bootstrap.initdb[+].waldir' "${PGHA_WALDIR}"
```

Comment:
This only has an effect when `initdb` is called. I wonder if it should move into the `"${PGHA_INIT}" == "true"` block above to be more clear?

Reply:
It would seem logical, IMO... :thinking:  :man_shrugging:

Reply:
@cbandy Can it only ever be called once and on init? Seemingly yes. Probably probably be moved then, of course with proper testing.

Reply:
This made me realize are prob some other configs that can now be left exclusively for the init config as well - essentially anything written to the `dcs` and/ or `initdb` sections. `initdb` is obviously only applicable at cluster creation, and all replicas inheriting DCS config from the actual DCS `configMap` once the cluster has been bootstrapped, so really only local config needs to be included when creating a new replica.

Very early on when designed as a standalone container, `crunchy-postgres-ha` was actually without the concept of an "init" node.  Basically, more than one node could be created at the same time when creating a new cluster, and all will fight to become leader and bootstrap the cluster.  This therefore meant that all those nodes would need the same init and DCS configs, in case they ultimately became responsible for bootstrapping the cluster.

However, as this container continues to evolve into what is essentially a PGO-exclusive container, (and since the Operator requires additional orchestration), these "standalone" container ideas and concepts are really no longer applicable, so we can rethink some of this design.  One thing that also comes to mind here are some of the places we set the `initdb` setting back to `true` (e.g. post-restore or for a cloned cluster), which would result in the execution of this logic, even though the `initdb` stuff isn't needed since the DB already exists (having it there is harmless, just something to consider cleaning up moving forward).

So anyway, just some food for thought moving forward since this got me thinking a bit.  For now I definitely think it makes sense to only include the WAL config when `PHGA_INIT` is `true`.